### PR TITLE
Add `aerius-file-server-client` as Maven package

### PR DESCRIPTION
**Pull request by wtenbosch**
Move the `fileserver-client` from the Register code to this repository, since it can also be used for other projects (at least Archive at this point). For project specific definition of the file structure, there is an interface. I've also already written the Register-implementation.

Added a root pom and made `fileserver` and `fileserver-client` childs of that. In theory we can now also share code between these two modules but I didn't make any effort right now. In the future, we might put the API spec for the fileserver in such a module? Or at least constants like the API method names and parameters.

### CODE REVIEW

**Review by Hilbrand — COMMENTED** [review]
By having multiple projects you might need to add a sonar profile and sonar-report module to allow sonar to collect coverage metrics. However since both projects are independent there are no cross-project metrics, therefor it might not be needed. We'll just see what happens in the sonar run and can add it later.

File: source/file-server/pom.xml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    <java.version>17</java.version>
+
+    <amazons3.version>2.20.56</amazons3.version>
+
+    <sonar.projectKey>aerius_file-server</sonar.projectKey>
+    <sonar.moduleKey>${project.artifactId}</sonar.moduleKey>
+    <sonar.organization>aerius</sonar.organization>
+    <sonar.host.url>https://sonarcloud.io</sonar.host.url>
+    <sonar.java.pmd.reportPaths>${project.build.directory}/dependency-check-report-pmd.xml</sonar.java.pmd.reportPaths>
+  </properties>
```
Comment (BertScholten) [inline source/file-server/pom.xml:43]: Don't think these are needed, as the parent pom specifies the same properties.

File: source/file-server/pom.xml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      <url>https://nexus.aerius.nl/repository/maven-public/</url>
+    </repository>
+  </repositories>
+  <pluginRepositories>
+    <pluginRepository>
+      <id>aerius-nexus-public</id>
+      <name>AERIUS Nexus Repository</name>
+      <url>https://nexus.aerius.nl/repository/maven-public/</url>
+    </pluginRepository>
+  </pluginRepositories>
```
Comment (BertScholten) [inline source/file-server/pom.xml:58]: Think you can remove this part as well, as it's already present in parent pom.

**Review by BertScholten — APPROVED** [review]
LGTM, just some parts that can be removed to keep things cleaner.

**Review by Hilbrand — APPROVED** [review]
LGTM

**Pull request merged**